src/base/bittorrent/filterparserthread.cpp

```diff
@@ -37,255 +42,151 @@
-            }
+        std::string_view range;
+        int accessLevel = 0;
+        // We don't care about this
+        // std::string_view description;
+    };
 
-            if (str != octetStart)
-            {
-                const long int extractedNum = strtol(octetStart, &endptr, 10);
-                if ((extractedNum >= 0L) && (extractedNum <= 255L))
-                    m_buf[octetIndex] = static_cast<unsigned char>(strtol(octetStart, &endptr, 10));
-                else
-                    return false;
-
-                if ((endptr == str) && (octetIndex == 3))
-                    return true;
-            }
+    IPFilterEntry parseDATLine(std::string_view line)
+    {
+        IPFilterEntry entry;
 
-            return false;
+        // Each line should follow this format:
+        // 001.009.096.105 - 001.009.096.105 , 000 , Some organization
+        // The 2nd entry is access level and if >=127 the IP range isn't blocked.
+        if (const auto firstComma = line.find(',');
+            firstComma != line.npos)
+        {
+            entry.range = line.substr(0, firstComma);
+
+            auto posIt  = std::find_if_not((line.cbegin() + firstComma + 1), line.cend(),
+                                          [](unsigned char c){ return std::isspace(c); } );
+            int pos = 0;
+            if (posIt != line.cend())
+                pos = posIt - line.cbegin();
+
+            // from_chars will stop when it encounters non-numeric chars
+            // we don't care for any other conversion failure
+            std::from_chars((line.data() + pos),
+                            (line.data() + line.size()),
+                            entry.accessLevel);
         }
-
-        lt::address_v4::bytes_type parsed() const
+        else
         {
-            return m_buf;
+            // Apparently only range is mandatory
+            entry.range = line;
         }
 
-    private:
-        lt::address_v4::bytes_type m_buf;
-    };
+        return entry;
+    }
 
-    bool parseIPAddress(const char *data, lt::address &address)
+    IPFilterEntry parseP2PLine(std::string_view line)
     {
-        IPv4Parser parser;
-        lt::error_code ec;
-
-        if (parser.tryParse(data))
-            address = lt::address_v4(parser.parsed());
+        IPFilterEntry entry;
+
+        // Each line should follow this format:
+        // Some organization:1.0.0.0-1.255.255.255
+        // The "Some organization" part might contain a ':' char itself so we find the last occurrence
+        // Apparently this format wasn't designed for IPv6, so no effort will be made
+        // to parse possible IPv6 literals wrapped in []. These characters can appear in
+        // the "some organization" part too. It isn't worth implementing such complex parsing
+        // for a format that doesn't support IPv6 and there aren't any lists with IPv6 in them.
+        if (const auto colon = line.rfind(':');
+            colon != line.npos)
```

Comment:
IMO putting it on the same line of `if` is easier to avoid tripping over when reading it (at least for me):
```c++ 
if (const auto colon = line.rfind(':'); colon != line.npos)
```